exist-core/src/test/xquery/xquery3/bang.xql

```diff
@@ -181,3 +181,46 @@ declare
 function bang:element-left-number-attribute-right() {
     <emp id='1'/> ! number(@id)
 };
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:map-get () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! map:get(., 'a')
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-get () {
+    ([1], [2], [3]) ! array:get(., 1)
+};
+
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:map-lookup () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! .?a
```

Comment:
this still fails

Reply:
Similarly to my comment above, would `?a` work, or is `.?` necessary to select the context item?

Reply:
This is a separate issue #1655 (see my comment below)

Reply:
Currently, one would need to write the above expression as
```xquery
(map {'a':1}, map {'a':2}, map {'a':3}) ! (.)?a
```
Working to resolve this as we are getting closer and closer to a stable solution

Reply:
done